src/Scheme/S7/s7.c

```diff
@@ -7208,8 +7208,17 @@ static void resize_heap_to(s7_scheme *sc, int64_t size)
       /* (sc->heap_size < 2048000) */  /* 8192000 here improves various gc benchmarks only slightly */
       /* maybe the choice of 4 should depend on how much space was freed rather than the current heap_size? */
       if (old_free < old_size * sc->gc_resize_heap_by_4_fraction)
-	sc->heap_size *= 4;          /* *8 if < 1M (or whatever) doesn't make much difference */
-      else sc->heap_size *= 2;
+	sc->heap_size *= 2;          /* *8 if < 1M (or whatever) doesn't make much difference */
```

Comment:
this part can be submitted to S7 like providing S7 internal options to control the behavior